dev/source/docs/copter-adding-custom-controller.rst

```diff
@@ -0,0 +1,314 @@
+.. _copter-adding-custom-controller:
+
+====================================================
+Adding Custom Attitude Controller to Copter
+====================================================
+
+Custom controller library allows you to implement and easily run your own controller inside ArduPilot in a systematic way. Original-primary-mains means existing controller and custom-secondary means new controller. This library aimed to not interfere with other parts of the main controller or vehicle level code. The controller output is sent to the control allocation library, known as the mixer, the same way the main controller does.
+The custom controller library has the following features:
+
+Features
+---------------------------------------------------
+
+- In-flight switching between main and custom controller with RC switch, option 109.
+- Bitmask to choose which axis to use the custom controller output
+- Filter, integrator reset mechanism when switching between controller   
+  - Bumpless transfer when switching from custom to the main controller
+- Ground and in-flight spool state separation to avoid build-up during arming and take-off with the custom controller
+- Frontend-backend separation that allows adding a new controller with very little overhead
+- Flag to compile out custom controller related code on hardware, --enable-custom-controller
+- Proper parameter table implementation that allows adding new custom controller parameter table without corruption
+- Single parameter to switch between different custom controllers, reboot required
+- Multiple checks to avoid accidentally running mis-un/configured custom controller with RC switch
+- Custom controller parameters start with CC_ in the GCS.
```

Comment:
```suggestion
- Custom controller parameters start with ``CC_`` in the GCS.
```
to avoid a build error...thinks its a param link

Reply:
Done.